files/nrpe/check_object.py

```diff
@@ -49,101 +45,56 @@ class OSCredentials(object):
   and provide the credentials.
   '''
   cred = dict()
-  keystone_cred = dict()
   keystone_v3_cred = dict()
 
   def __init__(self, options):
     self.environment_credentials()
     self.options_credentials(options)
-    self.credentials_available()
 
   def environment_credentials(self):
     try:
-      self.cred['auth_url']   = os.environ['OS_AUTH_URL']
-      self.cred['username']   = os.environ['OS_USERNAME']
-      self.cred['api_key']    = os.environ['OS_PASSWORD']
-      self.cred['project_id'] = os.environ['OS_TENANT_NAME']
-      # Keystone only entries
-      self.keystone_cred['auth_url']    = os.environ['OS_AUTH_URL']
-      self.keystone_cred['username']    = os.environ['OS_USERNAME']
-      self.keystone_cred['password']    = os.environ['OS_PASSWORD']
-      self.keystone_cred['tenant_name'] = os.environ['OS_TENANT_NAME']
       # Keystone v3 only entries
       self.keystone_v3_cred['auth_url']    = os.environ['OS_AUTH_URL']
       self.keystone_v3_cred['user_id']    = os.environ['OS_USERNAME']
       self.keystone_v3_cred['password']    = os.environ['OS_PASSWORD']
       self.keystone_v3_cred['project_id'] = os.environ['OS_TENANT_NAME']
+      self.keystone_v3_cred['user_domain_name'] = os.environ['OS_USER_DOMAIN_NAME']
+      self.keystone_v3_cred['project_domain_name'] = os.environ['OS_PROJECT_DOMAIN_NAME']
     except KeyError:
       pass
 
   def options_credentials(self, options):
-    if options.auth_url: self.cred['auth_url']   = options.auth_url
-    if options.username: self.cred['username']   = options.username
-    if options.password: self.cred['api_key']    = options.password
-    if options.tenant  : self.cred['project_id'] = options.tenant
-    # Keystone only entries
-    if options.auth_url: self.keystone_cred['auth_url']    = options.auth_url
-    if options.username: self.keystone_cred['username']    = options.username
-    if options.password: self.keystone_cred['password']    = options.password
-    if options.tenant  : self.keystone_cred['tenant_name'] = options.tenant
-    # Keystone v3 only entries
+    options.user_domain_name = 'Default'
     if options.auth_url: self.keystone_v3_cred['auth_url']    = options.auth_url
-    if options.username: self.keystone_v3_cred['user_id']    = options.username
+    if options.username: self.keystone_v3_cred['username']    = options.username
     if options.password: self.keystone_v3_cred['password']    = options.password
-    if options.tenant  : self.keystone_v3_cred['project_id'] = options.tenant
+    if options.tenant  : self.keystone_v3_cred['project_name'] = options.tenant
+    if options.user_domain_name: self.keystone_v3_cred['user_domain_name'] = options.user_domain_name
+    if options.user_domain_name: self.keystone_v3_cred['project_domain_name'] = options.user_domain_name
```

Comment:
This twisted my brain a bit, but comparing to the logic in check_openstack, it seems valid. In that other script the 2x assignments are combined into the same `if` block, I would have preferred the same style here. But more matter of taste

Reply:
Yeah, If I have to choose, I think I prefer the style in check_openstack.py. I think my reasoning for `check_openstack` was that I wanted to emphasis that "this is not like everything else" since the check_openstack is such a big script, while in check_object I did not think/ I did not consider it worth thinking about.